crates/net/network-api/src/lib.rs

```diff
@@ -67,8 +67,8 @@ pub trait PeersInfo: Send + Sync {
 /// Provides an API for managing the peers of the network.
 pub trait Peers: PeersInfo {
     /// Adds a peer to the peer set.
-    fn add_peer(&self, peer: PeerId, addr: SocketAddr) {
-        self.add_peer_kind(peer, PeerKind::Basic, addr);
+    fn add_peer(&self, peer: PeerId, tcp_addr: SocketAddr, udp_addr: Option<SocketAddr>) {
```

Comment:
these changes are a bit invasive, these changes can be prevented by adding another set of functions that also accept the `udp: SocketAddr` which then also delegate to add_peer_kind

Reply:
thanks for the tips, this really helpful, no need to adjust many test files.